### PR TITLE
`azurerm_kubernetes_cluster` - Updates no longer fail when using managed AAD integration

### LINKED ISSUE #7325: Updates to azurerm_kubernetes_cluster fail when cluster uses managed AAD integration
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.12.26
+ provider.azurerm v2.14.0

### Affected Resource(s)

* `azurerm_kubernetes_cluster`

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "aks" {
  name     = "aks-service-rg"
  location = "northeurope"
}

resource "azurerm_kubernetes_cluster" "aks" {
  name                            = "aks-service"
  location                        = azurerm_resource_group.aks.location
  resource_group_name             = azurerm_resource_group.aks.name
  node_resource_group             = "aks-infra-rg"
  dns_prefix                      = "aks-dev"
  enable_pod_security_policy      = false
  private_cluster_enabled         = false
  api_server_authorized_ip_ranges = null
 
  default_node_pool {
    name            = "default"
    node_count      = 4
    vm_size         = "Standard_B2ms"
    os_disk_size_gb = 30
    vnet_subnet_id  = var.virtual_network.subnets.aks.id
    max_pods        = 60
    type            = "VirtualMachineScaleSets"
  }

  linux_profile {
    admin_username = var.admin_username

    ssh_key {
      key_data = tls_private_key.aks.public_key_openssh
    }
  }
  
  role_based_access_control {
    enabled = true

    azure_active_directory {
      managed                 = true
      admin_group_object_ids  = [for key, value in local.cluster_admins : value.object_id] 
    }
  }

  identity {
    type    = "SystemAssigned"
  }

  addon_profile {

    azure_policy {
      enabled = true
    }
    
    oms_agent {
      enabled                    = true
      log_analytics_workspace_id = var.log_analytics_workspace.id
    }

    kube_dashboard {
      enabled = true
    }

    http_application_routing {
      enabled = false
    }

  }

  network_profile {
    network_plugin     = "azure"
    network_policy     = "azure"
    load_balancer_sku  = "Basic"
    service_cidr       = var.kubernetes_service_cidr
    docker_bridge_cidr = var.docker_bridge_cidr
    dns_service_ip     = cidrhost(var.kubernetes_service_cidr, 2)
  }

  tags = local.tags

}
```

### Expected Behavior
- Enable feature 'Microsoft.ContainerService/AAD-V2' on subscription
- Apply plan to create cluster with managed Azure Active Directory integration
- Change value of tags - or any other argument that doesn't necessitate a replacement of the resource
- Run terraform plan
- Apply plan
- Tags are updated to reflect changes

### Actual Behavior
- Enable feature 'Microsoft.ContainerService/AAD-V2' on subscription
- Apply plan to create cluster with managed Azure Active Directory integration
- Change value of tags - or any other argument that doesn't necessitate a replacement of the resource
- Run terraform plan
- Apply plan
- Apply fails with error: -

Error: updating Managed Kubernetes Cluster AAD Profile in cluster "aks-service" (Resource Group "aks-service-rg"): containerservice.ManagedClustersClient#ResetAADProfile: Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest" Message="Operation 'resetAADProfile' is not allowed for managed AAD enabled cluster."

### Steps to Reproduce
1. Register feature 'Microsoft.ContainerService/AAD-V2' on subscription as per https://docs.microsoft.com/en-us/azure/aks/managed-aad
2. `terraform plan`
3. `terraform apply`
4. Make changes to resource
5. `terraform plan`
6. `terraform apply`

### References

* #0000

### PR DESCRIPTION
**Pull request by aristosvo**
Fixes #7325

## Acceptance Tests:
```bash
make acctests SERVICE='containers' TESTARGS='-run=TestAccAzureRMKubernetesCluster_roleBasedAccessControlAAD'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./azurerm/internal/services/containers/tests/ -run=TestAccAzureRMKubernetesCluster_roleBasedAccessControlAAD -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccAzureRMKubernetesCluster_roleBasedAccessControlAAD
=== PAUSE TestAccAzureRMKubernetesCluster_roleBasedAccessControlAAD
=== RUN   TestAccAzureRMKubernetesCluster_roleBasedAccessControlAADUpdateToManaged
=== PAUSE TestAccAzureRMKubernetesCluster_roleBasedAccessControlAADUpdateToManaged
=== RUN   TestAccAzureRMKubernetesCluster_roleBasedAccessControlAADManaged
=== PAUSE TestAccAzureRMKubernetesCluster_roleBasedAccessControlAADManaged
=== RUN   TestAccAzureRMKubernetesCluster_roleBasedAccessControlAADManagedChange
=== PAUSE TestAccAzureRMKubernetesCluster_roleBasedAccessControlAADManagedChange
=== CONT  TestAccAzureRMKubernetesCluster_roleBasedAccessControlAAD
=== CONT  TestAccAzureRMKubernetesCluster_roleBasedAccessControlAADManagedChange
=== CONT  TestAccAzureRMKubernetesCluster_roleBasedAccessControlAADManaged
=== CONT  TestAccAzureRMKubernetesCluster_roleBasedAccessControlAADUpdateToManaged
--- PASS: TestAccAzureRMKubernetesCluster_roleBasedAccessControlAADManaged (720.09s)
--- PASS: TestAccAzureRMKubernetesCluster_roleBasedAccessControlAAD (776.63s)
--- PASS: TestAccAzureRMKubernetesCluster_roleBasedAccessControlAADUpdateToManaged (853.89s)
--- PASS: TestAccAzureRMKubernetesCluster_roleBasedAccessControlAADManagedChange (979.83s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/containers/tests      981.018s
```

### CODE REVIEW

File: azurerm/internal/services/containers/kubernetes_cluster_resource.go [hunk 1]
```diff
@@ -884,7 +884,7 @@ func resourceArmKubernetesClusterUpdate(d *schema.ResourceData, meta interface{}
 		}
 
 		// changing rbacEnabled must still force cluster recreation
-		if *props.EnableRBAC == rbacEnabled {
+		if *props.EnableRBAC == rbacEnabled && !*props.AadProfile.Managed {
```
Comment (mbfrahry) [inline azurerm/internal/services/containers/kubernetes_cluster_resource.go:887]: There is a potential crash here so we'll want to nil check this before pointing to it

File: azurerm/internal/services/containers/kubernetes_cluster_resource.go [hunk 2]
```diff
@@ -900,6 +900,11 @@ func resourceArmKubernetesClusterUpdate(d *schema.ResourceData, meta interface{}
 		} else {
 			updateCluster = true
 		}
+
+		if *props.AadProfile.Managed {
```
Comment (mbfrahry) [inline azurerm/internal/services/containers/kubernetes_cluster_resource.go:904]: And here as well

File: azurerm/internal/services/containers/kubernetes_cluster_resource.go (hunk 1 above)
Comment (aristosvo) [reply azurerm/internal/services/containers/kubernetes_cluster_resource.go]: Tried to resolve in a good way, but my Go experience is too limited to know what the best way is

File: azurerm/internal/services/containers/kubernetes_cluster_resource.go (hunk 2 above)
Comment (aristosvo) [reply azurerm/internal/services/containers/kubernetes_cluster_resource.go]: Same here

**Comment by aristosvo** [comment]
> Hey @aristosvo, these changes seem sane but we should add some nil checks and a test to confirm it works!

I've added tests to confirm that the following scenario's are working:
- Update from AAD-v1 to AAD-v2/Managed
- Node pool change with AAD-v2/Managed

**Pull request merged**